packages/overlay/test/index.ts

```diff
@@ -680,6 +680,39 @@ export const runOverlayTriggerTests = (type: string): void => {
                     'hover content is not available at point'
                 ).to.be.false;
             });
+
+            it('blocks body scroll when modal overlay is opened and restores when closed', async function () {
```

Comment:
nit: could we also add a behavioral test that verifies the page actually can’t scroll while the modal is open (and can again after it closes)? Right now we’re checking the body overflow style, which is fine, but it’s still an implementation detail. I was thinking we could create a long enough page, attempt a scroll (e.g. `window.scrollTo`), and assert on `window.scrollY`.

Reply:
Certainly we can. Great call. We haven't explored these tests yet on our patterns but would love to add these for others too.